rfcs/20190802-model-garden-redesign.md

```diff
@@ -0,0 +1,207 @@
+        modeling code, but have separate *main* modules. The main
+        module for benchmark testing will have addtional flags and setups for
+        performance testing.
+
+The following table shows the detailed view of proposed model directory
+structure. The SOTA model list will be updated to cover more categories.
+
+| Directory     | Subdirectories       |   | Explainations                   |
+:-------------- |:---------------------|:--|:------------------------------ |
+| modeling      |                      |   | Common modeling libraries       |
+|               | layers               |   | Common modules/layers, not built-in tensorflow layers yet.|
+|               | networks             |   | Well-known networks built on top of layers, e.g. transformer |
+|               | optimziers           |   | New or customized optimizers    |
+|               | training             |   | Training utils, e.g. example custom training loop |
+| utils         |                      |   | Miscellaneous Utilities         |
+|               | hyperparameters      |   | Common flags and model parameters. |
+|               | ...                  |   |                                 |
+| benchmarks    |                      |   | benchmark testing and reference models to validate tensorflow |                                |
+|               | utils                |   |                                 |
+|               | examples             |   | reference models for testing/validating end-to-end tensforflow |
+|               |                      | Resnet |                            |
+|               |                      | BERT   |                            |
+|               |                      | Transformer |                       |
+|               |                      | NCF   |                             |
+| nlp           |                      |   | models/tasks for Natural Language Processing |
+|               | utils                |   | NLP specific utils, e.g. input dataset |
+|               | BERT                 |   | NLP specific utils, e.g. input dataset |
+|               |                      | BERT core modeling |  |
+|               |                      | tasks | specific tasks on open public datasets, e.g Squad, MNLI |
+|               | XLNET                |    |   |
+|               | GPT                  |    |   |
+|               | Transformer          |    |   |
+|               | GNMT                 |    |   |
+|               | ...                  |    |   |
+| vision        |                      |    | models/tasks for Computer Vision |
+|               | image_classification |    |   |
+|               |                      | resnet |   |
+|               |                      | EfficientNet   |   |
+|               |                      | MnasNet   |   |
+|               |                      | ...   |   |
+|               | object_detection     |    |   |
```

Comment:
Unet is mainly used for segmentation. So instead of `object_detection`, rather make two sub-directories `object_detection` and `segmentation`. `segmentation` will then contain:
* `DeepLab v3/v3+`
* `Unet`
* `FastSCNN`
...

Reply:
Thanks! Updated to reflect the suggestion. We are still debating the best structure for vision models, this may subject to changes in near future.